app/shared/cms/haldoc.ts

```diff
@@ -1,98 +1,165 @@
-import {Http, Headers} from 'angular2/http';
 import {Observable} from 'rxjs/Observable';
-import TemplateParser from 'url-template';
+import {HalRemote} from './halremote';
+
+// Bring observables up to snuff
+export interface HalObservable<T> extends Observable<T> {
+  follow(rel: string, params?: {}): HalObservable<T>;
+  followList(rel: string, params?: {}): HalObservable<T[]>;
+  followItems(rel: string, params?: {}): HalObservable<T[]>;
+}
+Observable.prototype['follow'] = function(rel: string, params: {} = null) {
```

Comment:
Yeah, it turned out that extending Observable was really tough in typescript.  (See [this stackoverflow](http://stackoverflow.com/questions/30494708/how-can-i-create-a-class-that-extends-rx-observable-in-typescript)).

But adding to the Observable prototype is what they actually recommend in the rxjs docs.  So I opted to do that, and then use that `HalObservable` interface, which "knows about" the new methods.